api/client/registry/login.go

```diff
@@ -52,7 +53,16 @@ func runLogin(dockerCli *client.DockerCli, opts loginOptions) error {
 
 	var serverAddress string
 	var isDefaultRegistry bool
-	if opts.serverAddress != "" {
+
+	// when a user supplied registry url is the default docker.io registry just ignore it
+	// without this check the login will succed, but the credentials will not be saved properly and can't be used
+	// handle urls without schema for backward compatibility and pass the integration tests
+	u, _ := url.Parse(opts.serverAddress)
+	if u.Host == "" {
```

Comment:
I think the title I put is a bit confusing , the real problem here is that when a user does 
docker login docker.io
docker login http://docker.io

or any other variation
the client will save the credentials but after that when you do 

docker login the saved credentials are not used as they are not saved for the correct domain 
(https://registry-1.docker.io/v2/)

the check I added is simply to ignore the domain name if it is docker.io

all we care about is if the user wants to login to the default docker hub and if yes than ignore the hostname all together

that is why I don't see any point to care about parse errors as these are handled in the following blocks.

this fix is only so that the credentials don't get saved under an incorrect domain format